students/brandon_nguyen/session06/mailroom4.py

```diff
@@ -65,21 +69,77 @@ def create_rpt():
     print("\n\n")
 
 
-def sort_sum():
+# Test1 DONE
+def sort_sum4_report(dict_db):
     # creating a new list with computed value for easy printing.
     """
-    This function returned a sorted list of by on order amount3
+    This function returned a sorted list of by on order amount
     """
     dblist = []
     dbSum = []
     # quick way to convert back to list to reuse existing code.
     # list comprehension
-    [dblist.append((i, j)) for i, j in donor_db.items()]
+    [dblist.append((i, j)) for i, j in dict_db.items()]
     [dbSum.append((name, [round(sum(donation), 2)], [len(donation)],
      [round(sum(donation)/len(donation), 2)]),) for name, donation in dblist]
     return sorted(dbSum, key=lambda donor: donor[1], reverse=True)
 
 
+# Test 2: business logic of updating the donor db DONE
+def update_donation(input_val, dict_db):
+    """
+    To use this function please pass in two argument:
+    input_val is a list with two set of value, donor_name, donation_value.
+    db_dict is a replication of donor_db
+    """
+    # existing people add to donation
+    if input_val[0] in dict_db:
+        new_donation = dict_db.get(input_val[0])
+        new_donation.append(input_val[1])
+        dict_db[input_val[0]] = new_donation
+    else:
+        dict_db.update({input_val[0]: [input_val[1]]})
```

Comment:
Not related to unit testing, but still related to dictionary coding style. Lines 96-101 could have been simplified to `dict_db[input_val[0]].append(input_val[1])` if `dict_db` is defined as `defaultdict(list)`.